src/test/scala/io/iohk/ethereum/blockchain/sync/regular/NewRegularSyncSpec.scala

```diff
@@ -0,0 +1,633 @@
+
+        override lazy val ledger: TestLedgerImpl = stub[TestLedgerImpl]
+
+        var blockFetcher: ActorRef = _
+
+        var importedNewBlock = false
+        var importedLastTestBlock = false
+        (ledger.resolveBranch _).when(*).returns(NewBetterBranch(Nil))
+        (ledger
+          .importBlock(_: Block)(_: ExecutionContext))
+          .when(*, *)
+          .onCall((block, _) => {
+            if (block == newBlock) {
+              importedNewBlock = true
+              Future.successful(BlockImportedToTop(List(BlockData(newBlock, Nil, Block.number(newBlock)))))
+            } else {
+              if (block == testBlocks.last) {
+                importedLastTestBlock = true
+              }
+              Future.successful(BlockImportedToTop(Nil))
+            }
+          })
+
+        peersClient.setAutoPilot(new PeersClientAutoPilot)
+
+        def waitForSubscription(): Unit = {
+          peerEventBus.expectMsgClass(classOf[Subscribe])
+          blockFetcher = peerEventBus.sender()
+        }
+
+        def sendLastTestBlockAsTop(): Unit = sendNewBlock(testBlocks.last)
+
+        def sendNewBlock(block: Block = newBlock, peer: Peer = defaultPeer): Unit =
+          blockFetcher ! MessageFromPeer(NewBlock(block, Block.number(block)), peer.id)
+
+        def goToTop(): Unit = {
+          regularSync ! RegularSync.Start
+
+          waitForSubscription()
+          sendLastTestBlockAsTop()
+
+          awaitCond(importedLastTestBlock)
+        }
+      }
+
+      "import received new block" in new OnTopFixture(testSystem) {
+        goToTop()
+
+        sendNewBlock()
+
+        awaitCond(importedNewBlock)
+      }
+      "broadcast imported block" in new OnTopFixture(testSystem) {
+        goToTop()
+
+        etcPeerManager.expectMsg(GetHandshakedPeers)
+        etcPeerManager.reply(HandshakedPeers(handshakedPeers))
+
+        sendNewBlock()
+
+        etcPeerManager.fishForSpecificMessageMatching() {
+          case EtcPeerManagerActor.SendMessage(message, _) =>
+            message.underlyingMsg match {
+              case NewBlock(block, _) if block == newBlock => true
+              case _ => false
+            }
+          case _ => false
+        }
+      }
+      "update ommers for imported block" in new OnTopFixture(testSystem) {
+        goToTop()
+
+        sendNewBlock()
+
+        ommersPool.expectMsg(RemoveOmmers(newBlock.header :: newBlock.body.uncleNodesList.toList))
+      }
+      "fetch hashes if received NewHashes message" in new OnTopFixture(testSystem) {
+        goToTop()
+
+        blockFetcher !
+          MessageFromPeer(NewBlockHashes(List(BlockHash(Block.hash(newBlock), Block.number(newBlock)))), defaultPeer.id)
+
+        peersClient.expectMsgPF() {
+          case PeersClient.Request(GetBlockHeaders(_, _, _, _), _, _) => true
+        }
+      }
+    }
+  }
+
+  abstract class Fixture(_system: ActorSystem)
```

Comment:
maybe you could extract this as `NewRegularSyncFixture` because this file is already big